### ISSUE TITLE
p-sidebar not removing mask from DOM

### ISSUE THREAD

**Issue by eduardoleopoldino** [primefaces/primeng#10229]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please demonstrate your case at stackblitz by using the issue template below. Issues without a test case have much less possibility to be reviewd in detail and assisted.

https://primefaces.org/primeng/showcase/#/sidebar

**Current behavior**
Clicking on the buttons inside the component, the modal is not removed from the DOM. Using the close icon or clicking on the modal itself, it works correctly.

**Expected behavior**
It should remove the component's mask when hiding

**Minimal reproduction of the problem with instructions**
- Open sidebar component
- Click on save/cancel buttons

**Comment by yigitfindikli** [comment]
Fixed v11.4.1

**Comment by jiangyh1024** [comment]
happened again in v12
